src/game_battle.cpp

```diff
@@ -368,14 +368,22 @@ bool Game_Battle::AreConditionsMet(const RPG::TroopPageCondition& condition) {
 }
 
 bool Game_Battle::UpdateEvents() {
-	if (Game_Battle::CheckWin() || Game_Battle::CheckLose()) {
+	const auto battle_end = Game_Battle::CheckWin() || Game_Battle::CheckLose();
+
+	// 2k3 battle interupts events immediately when battle end conditions occur.
+	if (Player::IsRPG2k3() && battle_end) {
```

Comment:
Not sure if this check is needed because we already do a check here in the interpreter after each command:

https://github.com/EasyRPG/Player/blob/master/src/game_interpreter.cpp#L423-L426

Reply:
It is, because the return value of this function is used to determine whether we process battle actions. The interpreter aborts if the win conditions are set, but `IsRunning()` would still be true and we'd end up in an infinite loop.